Quickly written "How to" section

petterik opened the pull request:
Fixes #8 

Not the best readme, but we can update as we change things.

petterik committed:
Quickly written How To section

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+
+Now we'll want to talk to Arcadia via nREPL. To do this:
+
+1. Clone this repository: git@github.com:eponai/tools.cursive-arcadia-repl.git
```
The URL is now rendered properly, only the first part. Perhaps explicitly link this instead?

petterik replied:
https://github.com/eponai/tools.cursive-arcadia-repl/pull/6/files#r285394099

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+Now we'll want to talk to Arcadia via nREPL. To do this:
+
+1. Clone this repository: git@github.com:eponai/tools.cursive-arcadia-repl.git
+2. cd tools.cursive-arcadia-repl
```
Want to highlight with `` that this is a command?

petterik replied:
I replaced this ordered list (1. 2. 3.) with a single code block. What do you think?

dnegma commented on README.md:
````diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+              arcadia             {:local/root "../Arcadia"}}}' > deps.edn
+```
+
+2. Open and import the project with Cursive at that root.
````
Unclear what this means, do you mean open Cursive? Import Arcadia? Isn't that already done in `deps.edn`?

petterik replied:
This is outdated and has been reworded.

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
 
-## How to use
-TBD
+### How to use Cursive with Arcadia
```
This section assumes we already have Clojure CLI, a Unity project, and Arcadia, so perhaps we can add a Prerequisites section with more details;
- Link to [Clojure Guide](https://clojure.org/guides/getting_started)
- Link to how to setup [Arcadia with Unity](https://github.com/arcadia-unity/Arcadia/wiki/Getting-Started)

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+
+In your Unity project with Arcadia, do:
+
+1. Create a new "regular" looking clojure project that has a deps.edn at its root. For example:
```
Perhaps better to "decide" how to create the project, anyone already familiar with Clojure would probably have their own way of doing things anyway.
```suggestion
1. Create a new deps.edn clojure project. For example:
```

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+In your Unity project with Arcadia, do:
+
+1. Create a new "regular" looking clojure project that has a deps.edn at its root. For example:
+
```
Perhaps we can explain what this is doing, and use proper syntax highlighting so it's a little clearer what is intended? E.g something like:

Create a `deps.edn` file with the following: 
```Clojure
{:deps {org.clojure/clojure {:mvn/version "1.10.0"}
        arcadia             {:local/root "../Arcadia"}}}
```

dnegma commented on README.md:
```diff
@@ -6,5 +6,36 @@ are not redirected but evaluated in the bridge as they require the JVM.
+mkdir <project-name>
+cd <project-name>
+mkdir src test target
+echo '{:deps {org.clojure/clojure {:mvn/version "1.10.0"}
```
Do we need to import this if we're working with Unity?

petterik replied:
We still want a clojure version defined which Cursive thinks we're going to use. It'll use this clojure version to evaluate the code we write.

For reasons mentioned in https://github.com/eponai/tools.cursive-arcadia-repl/pull/5#discussion_r285393034 I still think we should **explicitly** define the clojure version used.

petterik committed:
Adds prerequisites section and re-arranges steps

petterik commented:
@dnegma Pushed new code. Changes:
* Added a prerequisites section.
* Re-arranged steps, so that one is going form "Unity specific" -> "Cursive specific"
* Added a testing section, to show how I have verified that the setup works.

dnegma commented on README.md:
```diff
@@ -6,5 +6,70 @@ are not redirected but evaluated in the bridge as they require the JVM.
+
+5. In Intellij/Cursive, open and import the `Assets/<project-name>` directory.
+
+6. Create a [remote REPL in Cursive](https://cursive-ide.com/userguide/repl.html#Remote_REPLs) on port `7888`.
```
```suggestion
6. Create a [remote REPL in Cursive](https://cursive-ide.com/userguide/repl.html#remote-repls) on port `7888`.
```

dnegma commented on README.md:
````diff
@@ -6,5 +6,70 @@ are not redirected but evaluated in the bridge as they require the JVM.
+        arcadia             {:local/root "../Arcadia"}}}
+```
+
+5. In Intellij/Cursive, open and import the `Assets/<project-name>` directory.
````
Does this need to be imported as a Clojure Deps project in IntelliJ?

petterik replied:
Yes, or configured afterwards if one doesn't do it.

@dnegma Do you think this is too unclear or can we merge this PR without this change?

dnegma replied:
We can merge this for now and add more changes later

petterik committed:
Fixes prerequisites section links and nested code block indentation

petterik committed:
Update README.md

Co-Authored-By: Diana Negma <dianagren@gmail.com>

petterik committed:
Adds step about adding deps.edn to Arcadia

petterik committed:
Adds the cursive-arcadia-repl as a dependency, instead of a clone + terminal command

petterik committed:
Removes the Arcadia deps.edn step, adds standalone process step

petterik committed:
Minor adjustments

petterik committed:
Wraps the deps.edn content in a markdown code block

petterik committed:
Renames step 3, about adding arcadia sources

petterik committed:
Removes whitespace

dnegma commented:
@petterik reminder commit prefix `docs:`
